youtube_dl/extractor/vidlii.py

```diff
@@ -0,0 +1,152 @@
+            'duration': 212,
+            'view_count': int,
+            'comment_count': int,
+            'average_rating': float,
+            'type': 'video',
+            'ext': 'mp4'
+        }
+    }, {
+        'url': 'https://www.vidlii.com/watch?v=vBo2IcrwOkO',
+        'md5': 'b42640a596b4dc986702567d49268963',
+        'info_dict': {
+            'id': 'vBo2IcrwOkO',
+            'title': '(OLD VIDEO) i like youtube!!',
+            'description': 'Original upload date:<br />\nMarch 10th 2011<br />\nCredit goes to people who own content in the video',
+            'thumbnail': 'https://www.vidlii.com/usfi/thmp/vBo2IcrwOkO.jpg',
+            'uploader': 'MyEditedVideoSpartan',
+            'url': 'https://cdn.vidlii.com/videos/vBo2IcrwOkO.mp4',
+            'uploader_url': 'https://www.vidlii.com/user/MyEditedVideoSpartan',
+            'upload_date': '20171011',
+            'categories': 'Film & Animation',
+            'tags': None,
+            'duration': 34,
+            'view_count': int,
+            'comment_count': int,
+            'average_rating': float,
+            'type': 'video',
+            'ext': 'mp4'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # extract basic properties of video
+        title = str_or_none(
+            self._html_search_regex(r'<h1>(.+?)</h1>', webpage,
+                                    'title', default=None)) or str_or_none(
+            self._html_search_regex(r'<title>([^<]+?)</title>', webpage,
+                                    'title', default=None)).replace(
+            " - VidLii", "") or str_or_none(
+            self._html_search_meta('twitter:title', webpage, 'title',
+                                   default=False)).replace(" - VidLii", "")
+
+        description = strip_or_none(
+            get_element_by_id('des_text', webpage).strip())
+
+        uploader = str_or_none(
+            self._html_search_regex(
+                r'<div[^>]+class="wt_person"[^>]*>(?:[^<]+)<a href="\/user\/[^>]*?>([^<]*?)<',
+                webpage,
+                'uploader', default=None)) or str_or_none(
+            self._html_search_regex(
+                r'<img src="[^>]+?class=["\']avt2\s*["\'][^>]+?alt=["\']([^"\']+?)["\']',
+                webpage, 'uploader', default=None))
+
+        url = self._html_search_regex(
+            r'videoInfo[\s]*=[\s]*{[^}]*src:[\s]*(?:"|\')([^"]*?)(?:"|\')',
+            webpage, 'url', default=None)
+
+        # get additional properties
+        uploader_url = "https://www.vidlii.com/user/%s" % uploader
+
+        # returns date as YYYYMMDD
+        upload_date = str_or_none(
+            self._html_search_meta('datePublished', webpage, 'upload_date',
+                                   default=False).replace("-",
+                                                          ""))
```

Comment:
Same.

Reply:
FIXED: Removed ```str_or_none(...)``` and replaced ```default=False``` by ```default=""``` such that ```replace``` can always take place.